### PR TITLE
[Process Agent] Move Windows Settings to Global Config

### PR DESCRIPTION
**Pull request by just-chillin**
### What does this PR do?

This PR removes unused settings from `WindowsConfig`. The remaining setting, `UsePerfCounters` is moved to the global config.

### Motivation

This PR is part of the process agent config refactor.

### Describe how to test/QA your changes

On Windows, set `process_config.windows.use_perf_counters` to `true`, as well as `log_level` to `info`. When you run the process check make sure you get the message "Using toolhelp API probe for process data collection".

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/process-agent-move-windows-settings-7c903ac4ba9fd393.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+enhancements:
+  - Added `DD_PROCESS_CONFIG_WINDOWS_USE_PERF_COUNTERS` and `DD_PROCESS_AGENT_WINDOWS_USE_PERF_COUNTERS` environment variables
+deprecations:
+  - Removed `process_config.windows.args_refresh_interval` config setting
+  - Removed `process_config.windows.add_new_args` config setting
```
Comment (xornivore) [inline releasenotes/notes/process-agent-move-windows-settings-7c903ac4ba9fd393.yaml:6]: Nit:
```suggestion
  - Removed unused `process_config.windows.args_refresh_interval` config setting
  - Removed unused `process_config.windows.add_new_args` config setting
```

**Review by xornivore — APPROVED** [review]
👍 left a minor edit for the release note.

File: releasenotes/notes/process-agent-move-windows-settings-7c903ac4ba9fd393.yaml (hunk 1 above)
Comment (just-chillin) [reply releasenotes/notes/process-agent-move-windows-settings-7c903ac4ba9fd393.yaml]: Great suggestion!

File: pkg/process/checks/process_windows_test.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+func TestPerfCountersConfigSetting(t *testing.T) {
+	cfg := config.Mock()
+	cfg.Set("process_config.windows.use_perf_counters", true)
```
Comment (mbotarro) [inline pkg/process/checks/process_windows_test.go:21]: nit: should we also test what's the default behavior?

**Review by mbotarro — APPROVED** [review]

**Review by alai97 — APPROVED** [review]

File: pkg/process/checks/process_windows_test.go (hunk 2 above)
Comment (just-chillin) [reply pkg/process/checks/process_windows_test.go]: Good idea, I added a test for it

File: pkg/process/checks/process_windows_test.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	}
+
+	t.Run("use toolhelp API", func(t *testing.T) {
+		resetOnce()
+		defer resetOnce()
+
+		cfg := config.Mock()
+		cfg.Set("process_config.windows.use_perf_counters", false)
+		probe := getProcessProbe()
+		assert.Equal(t, probe, defaultWindowsProbe)
```
Comment (mbotarro) [inline pkg/process/checks/process_windows_test.go:32]: nit: you can probably assert that you're able to cast `probe` to `windowsToolhelpProbe` in the first test case 
https://github.com/DataDog/datadog-agent/blob/b5bed4dc6a5b08a50346e614b386ea67e37695de/pkg/process/procutil/process_windows_toolhelp.go#L66-L68

and to `probe` in the second one
https://github.com/DataDog/datadog-agent/blob/b5bed4dc6a5b08a50346e614b386ea67e37695de/pkg/process/procutil/process_windows.go#L49-L58

**Review by djmitche — APPROVED** [review]

File: pkg/process/checks/process_windows_test.go (hunk 3 above)
Comment (just-chillin) [reply pkg/process/checks/process_windows_test.go]: Good idea. I used `assert.IsType` to fix it

**Pull request merged**